x-pack/legacy/plugins/apm/public/components/app/Settings/SettingsList.tsx

```diff
@@ -0,0 +1,297 @@
+import { ITableColumn, ManagedTable } from '../../shared/ManagedTable';
+import { CMListAPIResponse } from '../../../../server/lib/settings/cm/list_configurations';
+import { AddSettingsFlyout } from './AddSettings/AddSettingFlyout';
+import { APMLink } from '../../shared/Links/APMLink';
+
+export type Config = CMListAPIResponse[0];
+
+export function SettingsList() {
+  const { data = [], refresh } = useFetcher(loadCMList, []);
+  const [selectedConfig, setSelectedConfig] = useState<Config | null>(null);
+  const [isFlyoutOpen, setIsFlyoutOpen] = useState(false);
+
+  const COLUMNS: Array<ITableColumn<Config>> = [
+    {
+      field: 'service.name',
+      name: i18n.translate(
+        'xpack.apm.settings.cm.configTable.serviceNameColumnLabel',
+        {
+          defaultMessage: 'Service name'
+        }
+      ),
+      sortable: true,
+      render: (_, config: Config) => (
+        <EuiButtonEmpty
+          size="s"
+          color="primary"
+          onClick={() => {
+            setSelectedConfig(config);
+            setIsFlyoutOpen(true);
+          }}
+        >
+          {config.service.name}
+        </EuiButtonEmpty>
+      )
+    },
+    {
+      field: 'service.environment',
+      name: i18n.translate(
+        'xpack.apm.settings.cm.configTable.environmentColumnLabel',
+        {
+          defaultMessage: 'Service environment'
+        }
+      ),
+      sortable: true,
+      render: (value: string) => value
+    },
+    {
+      field: 'settings.sample_rate',
+      name: i18n.translate(
+        'xpack.apm.settings.cm.configTable.sampleRateColumnLabel',
+        {
+          defaultMessage: 'Sample rate'
+        }
+      ),
+      sortable: true,
+      render: (value: string) => value
+    },
+    {
+      field: 'timestamp.us',
+      name: i18n.translate(
+        'xpack.apm.settings.cm.configTable.lastUpdatedColumnLabel',
+        {
+          defaultMessage: 'Last updated'
+        }
+      ),
+      sortable: true,
+      render: (value: number) => (value ? moment(value / 1000).fromNow() : null)
+    },
+    {
+      name: '',
+      actions: [
+        {
+          name: i18n.translate(
+            'xpack.apm.settings.cm.configTable.editButtonLabel',
+            {
+              defaultMessage: 'Edit'
+            }
+          ),
+          description: i18n.translate(
+            'xpack.apm.settings.cm.configTable.editButtonDescription',
+            {
+              defaultMessage: 'Edit this config'
+            }
+          ),
+          icon: 'pencil',
+          color: 'primary',
+          type: 'icon',
+          onClick: (config: Config) => {
+            setSelectedConfig(config);
+            setIsFlyoutOpen(true);
+          }
+        }
+      ]
+    }
+  ];
+
+  const RETURN_TO_OVERVIEW_LINK_LABEL = i18n.translate(
+    'xpack.apm.settings.cm.returnToOverviewLinkLabel',
+    {
+      defaultMessage: 'Return to overview'
+    }
+  );
+
+  const hasConfigurations = data.length !== 0;
```

Comment:
Could `data.length` conceivably be `undefined`? In that case `data.length > 0` would be better. Or `!isEmpty(data)` which is more readable imo.